pyiceberg/table/update/snapshot.py

```diff
@@ -251,6 +253,13 @@ def _commit(self) -> UpdatesAndRequirements:
         )
         location_provider = self._transaction._table.location_provider()
         manifest_list_file_path = location_provider.new_metadata_location(file_name)
+
+        # get current snapshot id and starting snapshot id, and validate that there are no conflicts
+        if self._transaction._table.__class__.__name__ != "StagedTable":
```

Comment:
```suggestion
        if not isinstance(self._transaction._table, StagedTable):
```

Reply:
@Fokko, I tried this however by importing `StagedTable` to this file I create a cyclical import dependency. Any recommendations on how to solve this? Need to check this since `StagedTable` class doesn't override the `refresh()` function.

Reply:
How about importing it in the function? That should break the cyclical import.

Reply:
```suggestion
        from pyiceberg.table import StagedTable
        if not isinstance(self._transaction._table, StagedTable):
```